sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2105,6 +2105,12 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
                   catalog, "table-valued functions")
               }
             }
+            if (u.isStreaming && !resolvedFunc.isStreaming) {
+              // If the function was marked as streaming in the SQL statement but it's determined
+              // through resolution that the underlying function does not actually produce a
+              // streaming relation, throw an unsupported streaming exception.
+              throw QueryCompilationErrors.unsupportedStreamingTVF(u.funcName)
```

Comment:
to pass the fully qualified function name, we should do `ident.toQualifiedNameParts(catalog)` (this needs `import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._`)